src/analysis_and_optimization/Optimize.ml

```diff
@@ -777,7 +778,7 @@ let dead_code_elimination (mir : Program.Typed.t) =
             && is_skip_break_continue body.pattern
           then Skip
           else For {loopvar; lower; upper; body}
-      | Block l ->
+      | Profile (_, l) | Block l ->
           let l' = List.filter ~f:(fun x -> x.pattern <> Skip) l in
           if List.length l' = 0 then Skip else Block l'
```

Comment:
So dead code elimination pass converts profile blocks to ordinary blocks. I guess that's one way to not worry about optimized code profiling the wrong part. Was that intentional?

I think eventually we want optimization to keep profile blocks and not move code across their boundaries. But we can leave the interaction of profiling and optimization unspecified until we get around to addressing #545

Reply:
Yeah, mostly intentional because I dont know how it would work honestly. With inlining etc. 

I can look into that but not sure there is an easy way.

Reply:
Also agree with doing this after 545.